scanner.go

```diff
@@ -75,24 +75,35 @@ func (s *scanner) fetch() ([]*pb.Result, error) {
 }
 
 func (s *scanner) peek() (*pb.Result, error) {
-	if len(s.results) == 0 {
-		var (
-			err error
-			rs  []*pb.Result
-		)
-		if s.closed {
-			// done scanning
-			return nil, io.EOF
-		}
+	if len(s.results) > 0 {
+		return s.results[0], nil
+	}
 
-		rs, err = s.fetch()
-		if err != nil {
-			return nil, err
-		}
+	if s.closed {
+		// done scanning
+		return nil, io.EOF
+	}
 
-		// fetch cannot return zero results
-		s.results = rs
+	if s.renewCancel != nil {
+		// About to send new Scan request to HBase, cancel our
+		// renewer.
+		s.renewCancel()
+		s.renewCancel = nil
+	}
```

Comment:
Should we swap the order of these two if clauses? to ensure we cancel the renewer before returning EOF

Reply:
Done.